gruntwork-io/terraform-google-ci
# WIP: Add Tests & CircleCI Support

This PR adds tests written using [Terratest](https://github.com/gruntwork-io/terratest) and support for executing them using [CircleCI](https://circleci.com).

## Test Logic: Cloud Source Repositories Cloud Build Example
1. Clone the [Docker Sample App](https://github.com/gruntwork-io/sample-app-docker) to its own folder inside the build container.
1. Add changes to the repo and commit them.
1. Deploy all of the resources using Terraform.
1. Add a Git remote that points to the Cloud Source Repository (Note: this will require either a private key or using `gcloud`)
1. Check that a build was triggered using the Cloud Build API and monitor it.
1. Ensure the build was successful (this takes into account deployment to the GKE cluster)
1. Clean up all of the resources using Terraform `destroy`.
1. Ensure all of the GCR images are removed.

## Test Logic: GitHub Cloud Build Example
1. Clone the [Docker Sample App](https://github.com/gruntwork-io/sample-app-docker) to its own folder inside the build container.
1. Add changes to the repo and commit them.
1. Deploy all of the resources using Terraform.
1. Add a Git remote that points to the GitHub repo (Note: this will require a GitHub private/deploy key)
1. Check that a build was triggered using the Cloud Build API and monitor it.
1. Ensure the build was successful (this takes into account deployment to the GKE cluster)
1. Clean up all of the resources using Terraform `destroy`.
1. Ensure all of the GCR images are removed.
